src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -63,22 +66,45 @@ public function process(ContainerBuilder $container)
         $definition = $container->findDefinition($this->dispatcherService);
 
         foreach ($container->findTaggedServiceIds($this->listenerTag, true) as $id => $events) {
+            $reflection = null;
             foreach ($events as $event) {
                 $priority = isset($event['priority']) ? $event['priority'] : 0;
 
                 if (!isset($event['event'])) {
-                    throw new InvalidArgumentException(sprintf('Service "%s" must define the "event" attribute on "%s" tags.', $id, $this->listenerTag));
+                    if (null === $reflection) {
+                        $class = $container->getDefinition($id)->getClass();
+                        if (!$reflection = $container->getReflectionClass($class)) {
+                            throw new InvalidArgumentException(sprintf('Class "%s" used for service "%s" cannot be found.', $class, $id));
+                        }
+                    }
+                    if (!$reflection->implementsInterface(EventListenerInterface::class)) {
+                        throw new InvalidArgumentException(sprintf('Service "%s" must define the "event" attribute on "%s" tags or implements the EventListenerInterface interface .', $id, $this->listenerTag));
```

Comment:
interface should have FQCN + double quotes I think
extra space also before final dot